### PR TITLE
Bloom

### PR DESCRIPTION
**Pull request by gmodarelli**
Hello, this PR adds a bloom post-processing effect to the game.
I've also added support for ImGUI (we build it but don't use it yet) and I've done another pass on our lighting (there were other bugs).

The bloom implementation is not perfect, but I think it's a mix of current lighting and models as well. For example, the effect looks really nice with a reference PBR model like the Damaged Helmet in the screenshot below.

Screenshot 2023-09-29 065914

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -39,6 +39,22 @@ pub fn build(b: *std.Build) void {
         .dependencies = &.{},
     }));
 
+    // Linking ImGUI
+    const abi = (std.zig.system.NativeTargetInfo.detect(target) catch unreachable).target.abi;
```
Comment (Srekel) [inline build.zig:43]: Do we need this if we use zgui? Or are we not using zgui?

Comment (gmodarelli) [reply build.zig:43]: We can't use zgui unless we write a new glfw_d3d12 backend for it

File: src/shaders/deferred_lighting.hlsl [hunk 2]
```diff
@@ -44,72 +41,83 @@ void csDeferredLighting(uint3 dispatch_id : SV_DispatchThreadID) {
...
-        material_properties.baseColor = gbuffer_0_sample.rgb;
-        material_properties.metalness = gbuffer_2_sample.g;
-        material_properties.emissive = gbuffer_2_sample.b * gbuffer_0_sample.rgb;
-        material_properties.roughness = gbuffer_2_sample.r;
-        material_properties.transmissivness = 0;
-        material_properties.reflectance = 0.5;
-        material_properties.opacity = gbuffer_0_sample.a;
+        float3 albedo = gbuffer_0_sample.rgb;
+        float metallic = gbuffer_2_sample.g;
+        float3 f0 = float3(0.04, 0.04, 0.04);
```
Comment (Srekel) [inline src/shaders/deferred_lighting.hlsl:53]: Is this some magical PBR constant? :)

Comment (gmodarelli) [reply src/shaders/deferred_lighting.hlsl:53]: Yes :)

**Pull request merged**